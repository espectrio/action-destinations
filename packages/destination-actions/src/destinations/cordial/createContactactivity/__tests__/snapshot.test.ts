import { createTestEvent, createTestIntegration } from '@segment/actions-core'
import { generateTestData } from '../../../../lib/test-data'
import destination from '../../index'
import nock from 'nock'

const testDestination = createTestIntegration(destination)
const actionSlug = 'createContactactivity'
const destinationSlug = 'Cordial'
const seedName = `${destinationSlug}#${actionSlug}`

describe(`Testing snapshot for ${destinationSlug}'s ${actionSlug} destination action:`, () => {
  afterEach(() => nock.cleanAll())
  it('required fields', async () => {
    const action = destination.actions[actionSlug]
    const [eventData, settingsData] = generateTestData(seedName, destination, action, true)

    nock(/.*/)
      .persist()
      .post(/\/.*\/createContactactivity/)
      .reply(200)

    const event = createTestEvent(eventData)

    const responses = await testDestination.testAction(actionSlug, {
      event: event,
      mapping: eventData,
      settings: settingsData,
      auth: undefined
    })

    const request = responses[0].request
    const rawBody = await request.text()

    try {
      const json = JSON.parse(rawBody)
      expect(json).toMatchSnapshot()
      return
    } catch (err) {
      expect(rawBody).toMatchSnapshot()
    }

    expect(request.headers).toMatchSnapshot()
  })

  it('all fields', async () => {
    const action = destination.actions[actionSlug]
    const [eventData, settingsData] = generateTestData(seedName, destination, action, false)

    nock(/.*/)
      .persist()
      .post(/\/.*\/createContactactivity/)
      .reply(200)

    const event = createTestEvent({
      anonymousId: 'a922ab56-80b7-40c5-be4d-e089026593cd',
      properties: eventData.properties,
      sentAt: '2022-04-04T13:08:53.205Z',
      timestamp: '2022-04-04T13:08:53.205Z'
    })

    const mapping = {
      userIdentities: { 'channels.email.address': 'contact@example.com' }
    }

    const responses = await testDestination.testAction(actionSlug, {
      event: event,
      mapping: mapping,
      settings: settingsData,
      useDefaultMappings: true,
      auth: undefined
    })

    const request = responses[0].request
    const rawBody = await request.text()

    try {
      const json = JSON.parse(rawBody)
      expect(json).toMatchSnapshot()
      return
    } catch (err) {
      expect(rawBody).toMatchSnapshot()
    }
  })
})
