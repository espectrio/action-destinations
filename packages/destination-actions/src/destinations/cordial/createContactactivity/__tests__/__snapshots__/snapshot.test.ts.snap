// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing snapshot for Cordial's createContactactivity destination action: all fields 1`] = `
Object {
  "action": "Test Event",
  "anonymousId": "a922ab56-80b7-40c5-be4d-e089026593cd",
  "context": Object {
    "ip": "8.8.8.8",
    "library": Object {
      "name": "analytics.js",
      "version": "2.11.1",
    },
    "locale": "en-US",
    "location": Object {
      "city": "San Francisco",
      "country": "United States",
      "latitude": 40.2964197,
      "longitude": -76.9411617,
      "speed": 0,
    },
    "page": Object {
      "path": "/academy/",
      "referrer": "",
      "search": "",
      "title": "Analytics Academy",
      "url": "https://segment.com/academy/",
    },
    "timezone": "Europe/Amsterdam",
    "userAgent": "Mozilla/5.0 (iPhone; CPU iPhone OS 9_1 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13B143 Safari/601.1",
  },
  "properties": Object {
    "testType": "%!C)I@n7Uu54P",
  },
  "time": "2022-04-04T13:08:53.205Z",
  "userId": "user1234",
  "userIdentities": Object {
    "channels.email.address": "contact@example.com",
  },
}
`;

exports[`Testing snapshot for Cordial's createContactactivity destination action: required fields 1`] = `
Object {
  "action": "%!C)I@n7Uu54P",
  "userIdentities": Object {
    "testType": "%!C)I@n7Uu54P",
  },
}
`;
